plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -329,6 +329,13 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
             return unsignedMapping;
         }
 
+        String jdbcTypeName = typeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new TrinoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+        if ("datetime".equalsIgnoreCase(jdbcTypeName)) {
+            TimestampType timestampType = createTimestampType(getTimestampPrecision(typeHandle.getRequiredColumnSize()));
+            return Optional.of(timestampColumnMapping(timestampType));
+        }
```

Comment:
I don't understand why logic is requried. The new test (TestMariaDbTypeMapping#testTimestamp) passes without this change.

Reply:
`java.sql.Types` does not have `datatime` type, due to that can't map the `datetime`  based on jdbcType like it was done for `timestamp` inside the switch case condition.  For `datatime` we have to check based on the `jdbcTypeName` from the `typeHandle`.

Reply:
Could you explain why the new test passes without this change?

Reply:
Yea, typeHandle.getJdbcType() (93) is same for `datetime` and `timestamp`, we do not need to have separate handle for DATETIME type, added a code comment only.